src/app/components/signup/signup.component.ts

```diff
@@ -67,7 +67,20 @@ export class SignupComponent {
   ): ValidationErrors | null => {
     const password = control.get('password');
     const confirmPassword = control.get('confirmPassword');
-    return password?.value !== confirmPassword?.value ? {passwordMismatch: true} : null;
```

Comment:
it will update the state for formgroup not the formcontrol, so we need to manually update the state for control

Reply:
Can you explain more details, for learning

any article link or explanation would be appreciated.